internal/primitives/core/hash/hash.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package hash
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// H256 is a fixed-size uninterpreted hash type with 32 bytes (256 bits) size.
+type H256 string
```

Comment:
I think we can use `[32]byte` and it will still be `comparable`

Reply:
I need it to satisfy `constraints.Ordered` in future work.